test/exe/main_common_test.cc

```diff
@@ -38,34 +38,9 @@ class MainCommonTest : public testing::TestWithParam<Network::Address::IpVersion
       : config_file_(TestEnvironment::temporaryFileSubstitute(
             "test/config/integration/google_com_proxy_port_0.v2.yaml", TestEnvironment::ParamMap(),
             TestEnvironment::PortMap(), GetParam())),
-        random_string_(fmt::format("{}", computeBaseId())),
-        argv_({"envoy-static", "--base-id", random_string_.c_str(), "-c", config_file_.c_str(),
-               nullptr}) {}
-
-  /**
-   * Computes a numeric ID to incorporate into the names of
-   * shared-memory segments and domain sockets, to help keep them
-   * distinct from other tests that might be running concurrently.
-   *
-   * The PID is needed to isolate namespaces between concurrent
-   * processes in CI. The random number generator is needed
-   * sequentially executed test methods fail with an error in
-   * bindDomainSocket if the same base-id is re-used.
-   *
-   * @return uint32_t a unique numeric ID based on the PID and a random number.
-   */
-  static uint32_t computeBaseId() {
-    Runtime::RandomGeneratorImpl random_generator_;
-    // Pick a prime number to give more of the 32-bits of entropy to the PID, and the
-    // remainder to the random number.
-    const uint32_t four_digit_prime = 7919;
-#ifdef WIN32
-    return ::GetCurrentProcessId() * four_digit_prime +
-           random_generator_.random() % four_digit_prime;
-#else
-    return getpid() * four_digit_prime + random_generator_.random() % four_digit_prime;
-#endif
-  }
+        base_id_(TestEnvironment::chooseBaseId(3000000)),
```

Comment:
From my quick read of this you have selected a range of base IDs in the 1M, 2M, 3M range for various tests. Can you add comments to explain why that is and how the allocations work? Maybe there should be a central comment somewhere and a ref to that comment from each of the relevant locations in code and scripts?

Reply:
So when runs-per-test is set and multiple tests are executed (e.g. `bazel test --runs_per_test=2 //test:foo_test //test:bar_test`), the same TEST_RANDOM_SEED gets reused across tests. Each instance of a particular test (say //test:foo_test) sees unique values, but those same seeds are used again for other tests (say //test:bar_test). If both use hot restart they interfere with each other. I added this range setting to make it impossible for these tests to collide.

I'll add a comment and after I figure out why the tests failed.

Reply:
Ok. Test is fixed and the base-id nonsense is better documented.